### PR TITLE
[MMI] Added confirm-remove-jwt component

### PR DESCRIPTION
**Pull request by albertolive**
## Explanation

Added confirm-remove-jwt component

https://consensyssoftware.atlassian.net/browse/MMI-2643

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue

### CODE REVIEW

**Comment by albertolive** [comment]
It is blocked by `CustodyAccountList` component. Do not merge until the component is merged to develop

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        labels: item.labels,
+        balance: accounts.find(
+          (acc) => acc.address.toLowerCase() === item.address.toLowerCase(),
+        )?.balance,
+        token:
+          item.authDetails?.token ??
+          item.authDetails?.jwt ??
+          item.authDetails?.refreshToken,
+      }))
+      .filter((acc) => acc.token.toLowerCase() === token.address.toLowerCase());
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:49]: Question: can we move this filter to line 32 to not look the array one extra time?

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    setTokenAccounts(tokens);
+  }, [accounts, custodyAccountDetails, token]);
+
+  const handleCancel = () => {
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:54]: Can you add test to this one?
also if there's no more actions to this method, you can directly pass in `hideModal();` to `onCancel` to save some declaration ~

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  };
+
+  const handleRemove = async () => {
+    try {
+      for (const account of tokenAccounts) {
+        await dispatch(removeAccount(account.address.toLowerCase()));
+      }
+      handleCancel();
+    } catch (error) {
+      console.error(error);
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:65]: Should this error be included in `dispatch`?

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      .filter((acc) => acc.token.toLowerCase() === token.address.toLowerCase());
+
+    setTokenAccounts(tokens);
+  }, [accounts, custodyAccountDetails, token]);
+
+  const handleCancel = () => {
+    hideModal();
+  };
+
+  const handleRemove = async () => {
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:58]: Can you add some tests to this one ? thanks :)

File: ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js [hunk 5]
```diff
@@ -79,7 +79,7 @@ const ConfirmAddCustodianToken = () => {
                 setShowMore(true);
               }}
             >
-              {t('ShowMore')}
+              {t('showMore')}
```
Comment (DDDDDanica) [inline ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js:82]: do we have corresponding text to this key?

Comment (albertolive) [reply ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js:82]: Yes, there was a typo problem

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js (hunk 1 above)
Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:49]: Done

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js (hunk 2 above)
Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:54]: Done

File: ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js (hunk 3 above)
Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt.modal/confirm-remove-jwt-modal.js:65]: Would you happen to have an example?

File: ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        const addressLower = address.toLowerCase();
+        const account = accounts.find(
+          ({ address: adressAcc }) => adressAcc.toLowerCase() === addressLower,
+        );
+        const balance = account && account.balance;
+        const getToken =
+          authDetails?.token ?? authDetails?.jwt ?? authDetails?.refreshToken;
+        return { address, name, labels, balance, token: getToken };
+      })
+      .filter(
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js:47]: We can have filter first and then map so we only modify the valid elements. WdYt?

Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js:47]: Unfortunately, no, we need to filter after mapping. First, we want to get all the tokens from the below example and then filter by the one we passed via props. ie:

```
const getToken =
          authDetails?.token ?? authDetails?.jwt ?? authDetails?.refreshToken;
        return { address, name, labels, balance, **token**: getToken };
```

Comment (DDDDDanica) [reply ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js:47]: Thanks for explaining. It’s clear 🙏🏻

Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js]: @DDDDDanica I could add first the filter and it worked with some changes. Let me know :)

**Review by NidhiKJha — APPROVED** [review]

**Review by zone-live — APPROVED** [review]

File: ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  }, [accounts, custodyAccountDetails, token]);
+
+  const handleRemove = async () => {
+    try {
+      for (const account of tokenAccounts) {
+        await dispatch(removeAccount(account.address.toLowerCase()));
+      }
+      hideModal();
+    } catch (error) {
+      console.error(error);
```
Comment (DDDDDanica) [inline ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js:66]: Hey buddy, regarding the error handling, I'm not 100% sure about what is our standard and best practice here. But i can refer to some other components and we can add `  captureException(error);` to make it a bit more comprehensive, wdyt?
eg: `ui/ducks/institutional/institutional.js`,
`ui/ducks/alerts/unconnected-account.js` 

or we can `log` error, eg: `ui/ducks/swaps/swaps.js`
```
log.error(
        'Failed to fetch Swaps feature flags and Swaps liveness, defaulting to false.',
        error,
      );
```

Comment (albertolive) [reply ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js:66]: Ok, I'll talk with the team and if needed, we will create another PR. Thanks :)

**Pull request merged**